connaisseur/validators/cosign/cosign_validator.py

```diff
@@ -169,6 +252,77 @@ def __get_envs(self):
             env["SSL_CERT_FILE"] = f"/app/certs/cosign/{self.name}.crt"
         return env
 
+    @staticmethod
+    def __apply_policy(vals: dict, threshold: int, required: list):
+        """
+        Validates the signature verification outcome against the policy for
+        threshold and required trust roots.
+
+        Raises an exception if not compliant.
+        """
+
+        # verify threshold
+        signed_digests = [k["digest"] for k in vals.values() if k["digest"] is not None]
+        # raise exception if the same digest does not appear 'threshold' times
+        if not len(set(signed_digests)) == 1 or not len(signed_digests) >= threshold:
```

Comment:
Under which circumstance could the first condition fail?

Reply:
would fail if the list of signed digests contains different digests or none or if the number of digests is less than threshold